.azure-pipelines/ultimate-pipeline.yml

```
@@ -1164,6 +1167,80 @@ stages:
       condition: succeededOrFailed()
       continueOnError: true
 
+- stage: static_analysis_tests_profiler
+  condition: >
+    and(
+      succeeded(), 
+      eq(variables['isBenchmarksOnlyBuild'], 'False'),
+      eq(dependencies.generate_variables.outputs['generate_variables_job.generate_variables_step.IsProfilerChanged'], 'True')
+    )
+  dependsOn: [master_commit_id, generate_variables]
+  variables:
+    masterCommitId: $[ stageDependencies.master_commit_id.fetch.outputs['set_sha.master']]
+  jobs:
+  - template: steps/update-github-status-jobs.yml
+    parameters:
+      jobs: [Win, Linux]
+
+  - job: Win
+    pool:
+      name: azure-windows-scale-set-3
+    timeoutInMinutes: 60 #default value
+    
+    steps:
+    - powershell: |
+        choco install cppcheck -y
+        Write-Host "##vso[task.setvariable variable=PATH;]${env:PATH};C:\Program Files\Cppcheck";
+        refreshenv
+      displayName: Install CppCheck
+
+    - template: steps/clone-repo.yml
+      parameters:
+        masterCommitId: $(masterCommitId)
+
+    - template: steps/install-latest-dotnet-sdk.yml
+
+    - script: tracer\build.cmd RunCppCheckProfiler RunClangTidyProfiler
+      displayName: Run CppCheck and Clang-Tidy on Profiler
+
+    - script: tracer\build.cmd CheckProfilerStaticAnalysisResults
+      displayName: Check static analyzers results for errors
+
+    - publish: profiler/build_data
+      displayName: Publish Static Analysis Results
+      artifact: _$(System.StageName)_$(Agent.JobName)_logs_$(System.JobAttempt)
+      condition: succeededOrFailed()
+      continueOnError: true
+
+  - job: Linux
+    timeoutInMinutes: 60 #default value
+    pool:
+      name: azure-linux-scale-set
+
+    steps:
+    - template: steps/clone-repo.yml
+      parameters:
+        masterCommitId: $(masterCommitId)
+    - template: steps/install-latest-dotnet-sdk.yml
+
+    - template: steps/run-in-docker.yml
+      parameters:
+        build: true
+        target: builder
+        command: "Clean RunCppCheckProfiler RunClangTidyProfiler"
```

nit: presumably don't need the Clean here
```suggestion
        command: "RunCppCheckProfiler RunClangTidyProfiler"
```